File: WebApplication/ClientApp/src/app/_orchestration/getBreadcrumb/breadcrumbRequestDto.ts
import { removeUndefinedKeyValues } from "src/app/_common/jsonHelpers"

export enum BreadcrumbNavigationLevel {
  All,
  State,
  Area,
  City,
  ArpOrRaceSeriesDashboard,
  RaceLeaderboard,
  CourseLeaderboard,
  Irp
}

export class BreadcrumbRequestDto {

  constructor(
    public breadcrumbNavigationLevel: BreadcrumbNavigationLevel,
    public searchTerm: string,
  ) {}

  public getAsParamsObject = () => {

    return removeUndefinedKeyValues({
      breadcrumbNavigationLevel: this.breadcrumbNavigationLevel,
      searchTerm: this.searchTerm,
    })
  }
}

File: Orchestration/GetBreadcrumb/BreadcrumbRequestDto.cs
namespace Orchestration.GetBreadcrumb;

public enum BreadcrumbNavigationLevel
{
	All,
	State,
	Area,
	City,
	ArpOrRaceSeriesDashboard,
	RaceLeaderboard,
	CourseLeaderboard,
	Irp
}

public record BreadcrumbRequestDto
{
	public BreadcrumbNavigationLevel BreadcrumbNavigationLevel { get; set; }

	public string SearchTerm { get; set; }
}
